marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/DeployHubAmpsCommand.java

```diff
@@ -31,11 +42,53 @@ public DeployHubAmpsCommand(HubConfig hubConfig) {
         this.hubConfig = hubConfig;
     }
 
+    @Override
+    public void execute(CommandContext context) {
+        String stagingModulesDatabaseName = hubConfig.getStagingAppConfig().getModulesDatabaseName();
+        ManageConfig manageConfig = context.getManageClient().getManageConfig();
+        String securityUsername = manageConfig.getSecurityUsername();
+        String securityPassword = manageConfig.getSecurityPassword();
+        DatabaseClient installerClient = DatabaseClientFactory.newClient(
+            hubConfig.getHost(),
+            8000,
+            "Security",
+            new DatabaseClientFactory.DigestAuthContext(securityUsername, securityPassword)
+        );
+        //new AmpsInstaller(securityStagingClient).installAmps(stagingModulesDatabaseName);
```

Comment:
@grechaw ,

I think here it is assumed that "App-Services" is set to "digest". In case if all servers were set to cert-auth or other authentication type, this may not work. I see the same in undo() method as well

Reply:
Yes, I know -- I was hoping I could ask you how to generalize.

Reply:
I think we can use a method like newInstallerClient() in DeployHubAmpsCommand but that would mean that 'sslContext' for stagingAppConfig  has to be set to the one obtained using 'admin' certificate (cert whose common name is 'admin' or the security user , user will be logged as the security user). Is that an acceptable change ?

```
    private DatabaseClient newInstallerClient() {
    	ManageConfig manageConfig = ((HubConfigImpl)hubConfig).getManageConfig();
    	AppConfig stagingAppConfig = hubConfig.getStagingAppConfig();
    	DatabaseClientConfig config = new DatabaseClientConfig(hubConfig.getHost(), 8000, manageConfig.getSecurityUsername(), manageConfig.getSecurityPassword());
	    config.setCertFile(stagingAppConfig.getAppServicesCertFile());
	    config.setCertPassword(stagingAppConfig.getAppServicesCertPassword());
	    config.setDatabase("Security");
	    config.setExternalName(stagingAppConfig.getAppServicesExternalName());
	    config.setSecurityContextType(stagingAppConfig.getAppServicesSecurityContextType());
	    config.setSslContext(stagingAppConfig.getAppServicesSslContext());
	    config.setSslHostnameVerifier(stagingAppConfig.getAppServicesSslHostnameVerifier());
	    config.setTrustManager(stagingAppConfig.getAppServicesTrustManager());
	    return configuredDatabaseClientFactory.newDatabaseClient(config);
    }
```

Reply:
thanks @srinathgit that's exactly what i needed.  I'll take it for a spin

Reply:
Ok ... One thing I noticed with using this newInstallerClient() method is that, if all the servers (App-Services, Manage and Admin) are set to cert-auth, setting stagingAppConfig 's 'sslContext' to the one obtained from certificate whose common name matches 'securityUser' results in LoadModulesCommand and LoadSchemasCommand (and maybe some other command ?)running as 'securityUser' and not as 'hub-admin-user' as intended.